### DISCUSSION TITLE
Error with command "train"

### DISCUSSION THREAD

**Discussion by dosamy** [explosion/spaCy#7920] [Help: Coding & Implementations] [answered]
I prepared data for training a NER model in Spanish. I followed all the steps for creating the configuration file and the data in spacy's binary format. 
When I run the command `python -m spacy train config.cfg --paths.train ./train.spacy --paths.dev ./dev.spacy`, I get this error: 
`ValueError: Attempt to change dimension 'nI' for model 'maxout' from 768 to 192`

I have 2 questions please:
How can I solve this error to start the training?
What is the reason of the error as I haven't changed any configurations.

## Your Environment

- Operating System: Ubuntu
- Python Version Used: Python 3.8
- spaCy Version Used: Spacy 3

**Comment by svlandeg** [comment, marked as answer]
Hi! Can you share your full config file?

**Reply by dosamy** [reply]
Yes sure. 
Please note that when I was initializing the configuration and filling it, there was an error with **MultihashEmbed.v2**:
` @architectures = "spacy.MultiHashEmbed.v2"`
This was solved by changing it to **v1** `@architectures = "spacy.MultiHashEmbed.v1"`
This is the only modification done to the config file.

Here is my configuration:

`[paths]
train = null
dev = null
vectors = null
init_tok2vec = null

[system]
gpu_allocator = null
seed = 0

[nlp]
lang = "es"
pipeline = ["tok2vec","ner"]
batch_size = 1000
disabled = []
before_creation = null
after_creation = null
after_pipeline_creation = null
tokenizer = {"@tokenizers":"spacy.Tokenizer.v1"}

[components]

[components.ner]
factory = "ner"
moves = null
update_with_oracle_cut_size = 100

[components.ner.model]
@architectures = "spacy.TransitionBasedParser.v2"
state_type = "ner"
extra_state_tokens = false
hidden_width = 64
maxout_pieces = 2
use_upper = true
nO = null

[components.ner.model.tok2vec]
@architectures = "spacy.Tok2VecListener.v1"
width = ${components.tok2vec.model.encode.width}
upstream = "*"

[components.tok2vec]
factory = "tok2vec"

[components.tok2vec.model]
@architectures = "spacy.Tok2Vec.v2"

[components.tok2vec.model.embed]
@architectures = "spacy.MultiHashEmbed.v1"
width = 64
attrs = ["NORM","PREFIX","SUFFIX","SHAPE"]
rows = [2000,1000,1000,1000]
include_static_vectors = true

[components.tok2vec.model.encode]
@architectures = "spacy.MaxoutWindowEncoder.v2"
width = 256
depth = 8
window_size = 1
maxout_pieces = 3

[corpora]

[corpora.dev]
@readers = "spacy.Corpus.v1"
path = ${paths.dev}
max_length = 0
gold_preproc = false
limit = 0
augmenter = null

[corpora.train]
@readers = "spacy.Corpus.v1"
path = ${paths.train}
max_length = 2000
gold_preproc = false
limit = 0
augmenter = null

[training]
dev_corpus = "corpora.dev"
train_corpus = "corpora.train"
seed = ${system.seed}
gpu_allocator = ${system.gpu_allocator}
dropout = 0.1
accumulate_gradient = 1
patience = 1600
max_epochs = 10
max_steps = 2000
eval_frequency = 10
frozen_components = []
before_to_disk = null

[training.batcher]
@batchers = "spacy.batch_by_words.v1"
discard_oversize = false
tolerance = 0.2
get_length = null

[training.batcher.size]
@schedules = "compounding.v1"
start = 100
stop = 1000
compound = 1.001
t = 0.0

[training.logger]
@loggers = "spacy.ConsoleLogger.v1"
progress_bar = false

[training.optimizer]
@optimizers = "Adam.v1"
beta1 = 0.9
beta2 = 0.999
L2_is_weight_decay = true
L2 = 0.01
grad_clip = 1.0
use_averages = false
eps = 0.00000001
learn_rate = 0.001

[training.score_weights]
ents_per_type = null
ents_f = 1.0
ents_p = 0.0
ents_r = 0.0

[pretraining]

[initialize]
vectors = "es_core_news_lg"
init_tok2vec = ${paths.init_tok2vec}
vocab_data = null
lookups = null
before_init = null
after_init = null

[initialize.components]

[initialize.tokenizer]`

**Reply by svlandeg** [reply]
Which exact version of spacy are you using? Any chance you could upgrade to 3.0.6 and try initializing the config again from scratch?

**Reply by dosamy** [reply]
Thank you so much, @svlandeg!! I was using spacy version 3.0.1.
I updated to 3.0.6 as you recommended and I initialized the configuration and now it works.
Thank you again for your help!
Best regards,

**Reply by svlandeg** [reply]
Oh great, happy to hear it! Thanks for reporting back :-)

**Reply by dosamy** [reply]
Hello again:
I have further questions concerning the training and the config file. I am not sure if I should start a new issue or where to put it. Apologies for this inconvenience.
**Question 1:**
I used to train a model before using the config file with a python script where I had "20 iterations/epochs" and "batch size=256". Now, with the config file, I do not know how to set these parameters to ensure I am doing the same as before. So, my question is:
 How to set the number of iterations and batch size in the config file?

**Question 2:**
For the Spanish language there is a transformer model, but no "NER" component. Can I train a "ner" model for Spanish with transformers or unless the ner component is released I cannot do this?

Thank you in advance for your help!

**Reply by polm** [reply]
**Question 1:** You can read about how to set max epochs and iterations in https://github.com/explosion/spaCy/discussions/7465. An epoch is typically many iterations. For the batch size, your config has a parameter `batch_size = 1000` that you can adjust.

**Question 2:** You can train your own NER component if you have training data. I think the Transformers models just omit NER models due to size concerns.

**Reply by adrianeboyd** [reply]
For the NER models, it's the training data, not the size. We only have NER components for the transformer models if we have NER annotation on the same dataset as tags+parses, since in our experience the models don't train well from partial annotation.

**Reply by dosamy** [reply]
Thank you very much for  the clear answers.
It is really very helpful.
Best regards,
